impl/src/test/groovy/com/okta/sdk/impl/oauth2/AccessTokenRetrieverServiceImplTest.groovy

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright 2020-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.sdk.impl.oauth2
+
+import com.okta.commons.http.config.BaseUrlResolver
+import com.okta.sdk.ds.RequestBuilder
+import com.okta.sdk.impl.Util
+import com.okta.sdk.impl.api.OAuth2ClientCredentialsResolver
+import com.okta.sdk.impl.client.BaseClient
+import com.okta.sdk.impl.config.ClientConfiguration
+import com.okta.sdk.impl.error.DefaultError
+import com.okta.sdk.resource.ResourceException
+import io.jsonwebtoken.Claims
+import io.jsonwebtoken.Jwts
+import org.testng.annotations.Test
+
+import java.security.KeyPair
+import java.security.KeyPairGenerator
+import java.security.PrivateKey
+
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.Matchers.is
+import static org.hamcrest.Matchers.notNullValue
+import static org.mockito.ArgumentMatchers.any
+import static org.mockito.ArgumentMatchers.anyString
+import static org.mockito.Mockito.when
+import static org.mockito.Mockito.verify
+import static org.mockito.Mockito.mock
+import static org.mockito.Mockito.times
+import static org.testng.Assert.assertEquals
+
+/**
+ *
+ * Test for {@link AccessTokenRetrieverServiceImpl} class
+ */
+class AccessTokenRetrieverServiceImplTest {
+
+    @Test
+    void testInstantiationWithNullClientConfig() {
+        Util.expect(IllegalArgumentException) {
+            new AccessTokenRetrieverServiceImpl(null)
+        }
+    }
+
+    @Test
+    void testGetPrivateKeyFromPem() {
+
+        PrivateKey generatedPrivateKey = generatePrivateKey("RSA", 2048)
```

Comment:
perf tip: generating keys is slow, you can speed up tests be generating the key once in the constructor (or equivalent "before" method)

This may not be something you want to do though depending on what you are testing (i.e. if you need different keys for most of your tests).  Feel free to ignore this one :D

Reply:
true, didn't worry much about it because its a test anyway.